### PR TITLE
Move surface orientation calculation from tracking.singleaxis to new function

### LINKED ISSUE #1471: Consider extracting the surface orientation calculation in pvlib.tracking.singleaxis() to its own function
**Is your feature request related to a problem? Please describe.**
The usual workflow for modeling single-axis tracking in pvlib is to treat tracker rotation (`tracker_theta`) as an unknown to be calculated from solar position and array geometry.  However, sometimes a user might have their own tracker rotations but not have the corresponding `surface_tilt` and `surface_azimuth` values.  Here are a few motivating examples:
- Using measured rotation angles
- Post-processing the output of `tracking.singleaxis()` to include wind stow events or tracker stalls
- Other tracking algorithms that determine rotation differently from the astronomical method

Assuming I have my tracker rotations already in hand, getting the corresponding `surface_tilt` and `surface_azimuth` angles is not as easy as it should be.  For the specific case of horizontal N-S axis the math isn't so bad, but either way it's annoying to have to DIY when pvlib already has code to calculate those angles from tracker rotation.

**Describe the solution you'd like**
A function `pvlib.tracking.rotation_to_orientation` that implements the same math in `pvlib.tracking.singleaxis` to go from `tracker_theta` to `surface_tilt` and `surface_azimuth`.  Basically extract out the second half of `tracking.singleaxis` into a new function.  Suggestions for the function name are welcome.  To be explicit, this is more or less what I'm imagining:

```python
def rotation_to_orientation(tracker_theta, axis_tilt=0, axis_azimuth=0, max_angle=90):
    # insert math from second half of tracking.singleaxis() here
    out = {'tracker_theta': tracker_theta, 'aoi': aoi,
           'surface_tilt': surface_tilt, 'surface_azimuth': surface_azimuth}
    return pandas_if_needed(out)
```

**Describe alternatives you've considered**
Continue suffering

**Additional context**
This is one step towards a broader goal I have for `pvlib.tracking` to house other methods to determine tracker rotation in addition to the current astronomical method, the same way we have multiple temperature and transposition models.  These functions would be responsible for determining tracker rotations, and they'd all use this `rotation_to_orientation` function to convert rotation to module orientation.

Separately, I wonder if the code could be simplified using the tilt and azimuth equations in Bill's technical report (https://www.nrel.gov/docs/fy13osti/58891.pdf) -- seems like what we're doing is overly complicated, although maybe I've just not studied it closely enough.

cc @williamhobbs @spaneja

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] Closes #1471
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/reference) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: pvlib/tracking.py [hunk 1]
```diff
@@ -446,88 +448,75 @@ def singleaxis(apparent_zenith, apparent_azimuth,
...
 
 
+def calc_surface_orientation(tracker_theta, axis_tilt=0, axis_azimuth=0):
+    """
+    Calculate the surface tilt and azimuth angles for a given tracker rotation.
+
+    Parameters
+    ----------
+    tracker_theta : numeric
+        Tracker rotation angle [degrees]
```
Comment (cwhanse) [inline pvlib/tracking.py:477]: I think this docstring needs to specify (somewhere) which rotation direction is negative.

File: pvlib/tracking.py [hunk 2]
```diff
@@ -446,88 +448,75 @@ def singleaxis(apparent_zenith, apparent_azimuth,
...
+    axis_tilt : float, default 0
+        The tilt of the axis of rotation with respect to horizontal [degrees]
```
Comment (cwhanse) [inline pvlib/tracking.py:479]:
```suggestion
        The tilt of the axis of rotation with respect to horizontal. [degree]
```

File: pvlib/tracking.py [hunk 3]
```diff
@@ -446,88 +448,75 @@ def singleaxis(apparent_zenith, apparent_azimuth,
...
+    axis_azimuth : float, default 0
+        A value denoting the compass direction along which the axis of
+        rotation lies. Measured east of north. [degrees]
+
+    Returns
+    -------
+    dict or DataFrame
+        Contains keys ``'surface_tilt'`` and ``'surface_azimuth'`` representing
+        the module orientation accounting for tracker rotation and axis
+        orientation. [degrees]
```
Comment (cwhanse) [inline pvlib/tracking.py:489]:
```suggestion
        orientation. [degree]
```

File: pvlib/tracking.py [hunk 4]
```diff
@@ -334,9 +334,9 @@ def singleaxis(apparent_zenith, apparent_azimuth,
     Returns
     -------
     dict or DataFrame with the following columns:
-        * `tracker_theta`: The rotation angle of the tracker.
-          tracker_theta = 0 is horizontal, and positive rotation angles are
-          clockwise. [degrees]
+        * `tracker_theta`: The rotation angle of the tracker as a right-handed
```
Comment (cwhanse) [inline pvlib/tracking.py:337]:
```suggestion
        * `tracker_theta`: The rotation angle of the tracker is a right-handed
```

File: pvlib/tracking.py [hunk 5]
```diff
@@ -446,88 +448,79 @@ def singleaxis(apparent_zenith, apparent_azimuth,
...
+    Returns
+    -------
+    dict or DataFrame
+        Contains keys ``'surface_tilt'`` and ``'surface_azimuth'`` representing
+        the module orientation accounting for tracker rotation and axis
+        orientation. [degree]
+
+    References
+    ----------
+    .. [1] William F Marion and Aron P Dobos, "Rotation Angle for the Optimum
```
Comment (cwhanse) [inline pvlib/tracking.py:497]:
```suggestion
    .. [1] William F. Marion and Aron P. Dobos, "Rotation Angle for the Optimum
```

**Review by cwhanse — APPROVED** [review]

**Comment by wholmgren** [comment]
As a follow up, should we remove https://github.com/pvlib/pvlib-python/blob/master/docs/tutorials/tracking.ipynb? Looks like a lot of work to bring that notebook up to date with the current implementation and the newer features.

I'd also recommend updating the PR title to reflect the use of new equations.

**Comment by kanderso-nrel** [comment]
> should we remove https://github.com/pvlib/pvlib-python/blob/master/docs/tutorials/tracking.ipynb? 

Ugh, maybe.  I hate to delete useful info, even if the specific implementation doesn't match the current pvlib.  If there is effort to spend, I think I'd rather put it into a new User Guide page to replace that notebook instead of updating the notebook itself.  Or maybe improved gallery examples as suggested in https://github.com/pvlib/pvlib-python/pull/1077#pullrequestreview-504939228.

> I'd also recommend updating the PR title to reflect the use of new equations.

Good point.  Wish I had done that before the title got used for the squashed commit message.  Oh well!

**Comment by wholmgren** [comment]
I agree that a user guide page and/or gallery examples are better than updating the notebook. 

I didn't pay much attention to this PR since the title suggested that it was just a matter of moving code from one function to another :)

**Comment by kanderso-nrel** [comment]
> I didn't pay much attention to this PR since the title suggested that it was just a matter of moving code from one function to another :)

Yep, that's my bad.  Please feel free to point out anything you wish had been done prior to merging and I will happily rectify it!

**Pull request merged**